src/lib/customPropTypes.js

```diff
@@ -181,6 +181,71 @@ export const demand = (requiredProps) => {
 }
 
 /**
+ * Node shorthand ensures a prop is a node and that it's not valid with children.
+ */
+export const nodeShorthand = (...args) => every([
+  disallow(['children']),
+  PropTypes.node,
+])(...args)
+
+/**
+ * Item shorthand is a description of a component that can be a literal,
+ * a props object, or an element.
+ */
+export const itemShorthand = (...args) => every([
+  disallow(['children']),
+  PropTypes.oneOfType([
+    PropTypes.element,
+    PropTypes.number,
+    PropTypes.object,
+    PropTypes.string,
+  ]),
+])(...args)
+
+/**
+ * ArrayOf shorthand ensures a prop is an array of itemShorthand
+ */
+export const arrayOfShorthand = (...args) => every([
+  disallow(['children']),
+  PropTypes.arrayOf(itemShorthand),
+])(...args)
+
+/**
+ * Collection shorthand is a description of a component that can be a literal,
+ * a props object, or an element.
+ */
+export const collectionShorthand = (...args) => every([
+  disallow(['children']),
+  PropTypes.oneOfType([
+    arrayOfShorthand,
+    PropTypes.element,
+    PropTypes.object,
+  ]),
+])(...args)
+
+const shorthandPropTypes = [
+  arrayOfShorthand,
+  collectionShorthand,
+  itemShorthand,
+  nodeShorthand,
+]
+
+/**
+ * Ensure a prop conforms to children prop standards.
+ */
+export const children = (Component) => (...args) => {
+  const disallowedProps = _.flow(
+    _.pickBy(checker => _.some(shorthandPropTypes, (propType) => propType === checker)),
```

Comment:
Is this working correctly?  This file uses `lodash/fp` which takes data last, so I'd expect the `shorthandPropTypes` (data) to be provided after the `some` function.

Also, might be slightly cleaner to use `_.includes`:

```jsx
_.pickBy(checker => _.includes(checker, shorthandPropTypes))
```
I believe this is correct, data last again.  Didn't test it though.